tests/FsCheck.Test/Runner.fs

```diff
@@ -1,5 +1,150 @@
+
+    [<Fact>]
+    let ``parallelTest produces same sequence as test on failure`` () =
+        let body i = i < 999
+        let arb = Arb.from<int>
+        let config = { Config.Quick with MaxTest = 30000; EndSize = 30000 }
+        if not <| check arb body config then failwith "can't find the way to call xunit assertion lol"
+            
+    [<Fact>]
+    let ``parallelTest produces same sequence as test on discard`` () =
+        let body _ = true
+        let arb = Arb.fromGen (Gen.constant 1 |> Gen.map (fun _ -> Prop.discard ()))
+        let config = { Config.Quick with MaxTest = 30000; MaxFail = 100000; EndSize = 30000 }
+        if not <| check arb body config then failwith "can't find the way to call xunit assertion lol"
+    
+    type Tree = Leaf of int | Branch of Tree * Tree
+    
+    let tree =
+        let rec tree' s = 
+            match s with
+            | 0 -> Gen.map Leaf Arb.generate<int>
+            | n when n>0 -> 
+                let subtree = tree' (n/2)
+                Gen.oneof [ Gen.map Leaf Arb.generate<int> 
+                            Gen.map2 (fun x y -> Branch (x,y)) subtree subtree]
+            | _ -> invalidArg "s" "Only positive arguments are allowed"
+        Gen.sized tree'
+        
+    type TreeGen =
+        static member Tree() =
+            {new Arbitrary<Tree>() with
+                override x.Generator = tree
+                override x.Shrinker t = Seq.empty }
+
+    [<Fact>]
+    let ``parallelTest works with custom arb`` () =
+        Arb.register<TreeGen> ()
+        let arb = Arb.from<list<Tree>>
+        let body (xs:list<Tree>) = List.rev(List.rev xs) = xs
+        let config = { Config.Quick with MaxTest = 3000; EndSize = 3000 }
+        if not <| check arb body config then failwith "not threadsafe"
+    
+    type Integer = Integer of int
+    
+    let integer =
+        let mutable i = 0
+        let rec integer' s = 
+            i <- i + s // breaking thread safety
+            Integer i |> gen.Return
+        Gen.sized integer'
+        
+    type IntegerGen =
+        static member Integer() =
+            {new Arbitrary<Integer>() with
+                override x.Generator = integer
+                override x.Shrinker t = Seq.empty }
+
+    [<Fact>]
+    let ``parallelTest doesnt works with custom non threadsafe arb`` () =
+        Arb.register<IntegerGen> ()
+        let body i = true
+        let arb = Arb.from<Integer>
+        let config = { Config.Quick with MaxTest = 300000; EndSize = 300000 }
+        if check arb body config then failwith "can't be"
+
+    [<Property(Replay = "10538531436017130025,14826463994991344553", MaxTest = 30000, EndSize = 30000)>]
+    let ``30k ints fails`` () =
+        true
+        
+    [<Fact>]
+    let ``bad case`` () = 
+        let rnd = Random.createWithSeedAndGamma (13471455474525100574UL,7555858534656909083UL)
+        let n = 19938
+        Random.rangeInt (0, n, rnd)
```

Comment:
What are these two tests checking for?

Reply:
At some moment my implementation was failing and these tests are repro for such case. We don't need it anymore, I will remove it.

Reply:
Oh I recalled these tests are for #356 
They already shipped in that PR so removing it here as duplicates.